hugo/content/docs/actions_and_protocols.md

```diff
@@ -0,0 +1,96 @@
+In this section, the most important actions for understanding the basic communication protocols are described.
+For more info on existing actions see [actions.ts](https://github.com/eclipse-sprotty/sprotty/blob/master/packages/sprotty-protocol/src/actions.ts)  
+### ComputedBoundsAction
+Sent from the client to the model source to transmit the result of bounds
+computation for micro-layout as a response to a RequestBoundsAction
+
+### RequestBoundsAction
+Sent from the model source to the client to request bounds for the given model. This triggers the micro-layout computation.
+### RequestModelAction
+This is sent from client to model source to request a new model. Usually, this is the first action sent to initiate the communication.
+optionally this can contain an `options` object containing configuration for the `DiagramServer`, like properties for `needsClientLayout` and `needsServerLayout`
+### SetModelAction
+This is sent from the model source to the client to set the model. It contains the schema for the new graph.
+Should a model already exist in the client, it is overwritten.
+### UpdateModelAction
+Sent from model source to client to update the current model. Allows animating the transition from the old to the new model and contains properties for that.
+
+## Protocols
+Based on where layouting should be done there are 3 different protocols through which the `DiagramServer` and client can communicate.
+On client-side, this can be configured through the `configureViewerOptions` in the dependency injection container.
+```Typescript
+configureViewerOptions(context, {
+    needsClientLayout: false,
+    needServerLayout: true,
+    baseDiv: containerId
+});
+```
+
+On server-side layouting options need to be set in the `DiagramServer`
+```Typescript
+diagramServer.setNeedsClientLayout(false);
+diagramServer.setNeedsServerLayout(true);
+```
+
+### 1. No Layout Computation / Server-Only Layout
+
+In this scenario, the server needs to provide a model with complete layout information, so that no further processing by the client is required.
+{{< mermaid class="text-center">}}
+sequenceDiagram
+participant C as Client
+participant S as Server
+Note over C,S: Client requests model
+C->>S: RequestModelAction
+S->>C: SetModelAction
+Note over C,S: Server updates model
+loop when model changes
+    S->>C: UpdateModelAction
+end
+{{< /mermaid>}}
+In this, first the client requests and receives the model from the server. After that, the server can update the model whenever necessary.
+
+### 2. Client-Only Layout
+
+The server sends RequestBoundsAction to the client instead of updating the model directly. The client does not forward the resulting ComputedBoundsAction to the server because that would be an unnecessary round-trip. The updated bounds are applied locally in the client instead.
+{{< mermaid class="text-center">}}
```

Comment:
its very similar but not identical. The returned types are `RequestBoundsAction` instead of `SetModelAction` and `UpdateModelAction`. but yeah the steps of communication are basicly the same.
I could of course remove the diagram and just reference the previous one but i like it this way it is right now

Reply:
Ah, I see it now, thanks for pointing that out. I would recommend to change the color of that item to make it a bit easier to see the diff between the two.

Reply:
I had the same problem. I could bet they are the same, but after reading @jonah-iden comment, I see the difference. It's like voodoo, really scary. 
I like @montymxb suggestion